File: Leetcode/src/com/leetcode/bitManipulation/_1310_XORQueriesofaSubarray.java
package com.leetcode.bitManipulation;

public class _1310_XORQueriesofaSubarray {
    public int[] xorQueries(int[] arr, int[][] queries) {
        int n = arr.length;
        int[] xorPrefix = new int[n + 1];
        for (int i = 1; i <= n; i++) {
            xorPrefix[i] = xorPrefix[i - 1] ^ arr[i - 1];
        }
        int[] res = new int[queries.length];
        for (int i = 0; i < queries.length; i++) {
            int[] query = queries[i];
            res[i] = xorPrefix[query[0]] ^ xorPrefix[query[1] + 1];
        }
        return res;
    }
}

File: Leetcode/src/com/leetcode/bit_manipulation/_1310_xor_queries_of_a_subarray.py
from typing import List


class Solution:
    def xorQueries(self, arr: List[int], queries: List[List[int]]) -> List[int]:
        n = len(arr)
        prefix = [0] * (n + 1)
        for i in range(1, n + 1):
            prefix[i] = arr[i - 1] ^ prefix[i - 1]

        res = [0] * len(queries)
        for i in range(len(queries)):
            s, e = queries[i]
            res[i] = prefix[s] ^ prefix[e + 1]

        return res
